probe-rs/src/architecture/arm/memory/adi_v5_memory_interface.rs

```diff
@@ -46,6 +46,13 @@ where
     // If it doesn't support it, bit 30 in the CSW register has
     // to be set to 1 at all times.
     supports_hnonsec: bool,
+
+    // Cached value of the CSW register, to avoid unecessary writes.
+    //
+    // TODO: This is the wrong location for this, it should actually be
```

Comment:
Did you want to move this first?

Reply:
I want to move it, but I didn't really find a convenient way of doing it :grimacing: 

This is more a reminder for later refactoring.

Reply:
Ok that's fine with me! Guess we can merge this then; I fear that someone introduced some merge conflicts with this PR in the meanwhile tho ;)